app/lib/dartdoc/backend.dart

```diff
@@ -320,3 +320,58 @@ class DartdocBackend {
     await deletePool.close();
   }
 }
+
+class ArchiveTracker {
+  final _pvMap = <String, _TrackedEntry>{};
+  final _sampleLines = <String>[];
+
+  ArchiveTracker() {
+    _logger.info('Tracking: init.');
+    new Timer.periodic(const Duration(minutes: 10), (_) => _cleanup());
+    new Timer.periodic(const Duration(minutes: 45), (_) => _logSample());
```

Comment:
Consider: There's no way to prevent these from being turned on immediately if something wanted to construct such an object but not use it yet (pass it to something else). There's no way to turn them off when the program is done using the object. This may well be fine and I do the same in some of my code. Since this class is constructor as global objects, there's no way to import the files in question without automatically starting these side effects, making it harder to reuse the code for testing and other purposes.

The cleanup should happen as part of the logging samples. That way the intervals of cleanup and logging, which doesn't align, doesn't interfere with each other.

Reply:
I thought that the cleanup should be more frequent, similar to what I'd do if there would be an actual disk-based cache. The log intervals are more wide, with the goal having a good granularity over a day's use, but not too detailed information.

Re: side effects and reusability: I'd throw this class out once we have the info. I see no reason to make it robust and testable, but I can certainly make it so if needed.